sqlmesh/core/scheduler.py

```diff
@@ -431,6 +438,37 @@ def compute_interval_params(
         ignore_cron=ignore_cron,
         end_bounded=end_bounded,
     ).items():
+
+        def intersect(left: SnapshotInterval, intervals: list[Interval]) -> SnapshotInterval | None:
+            """Returns the intersection of `left` against `intervals`. Returns None if it does not intersect. Assumes `intervals` are disjoint."""
+
+            left_lower, left_upper = left
+            for right in intervals:
+                right_lower, right_upper = to_timestamp(right[0]), to_timestamp(right[1])
+                lower = max(left_lower, right_lower)
+                upper = min(left_upper, right_upper)
+                if lower <= upper:
+                    return (lower, upper)
+            return None
+
+        if signal_factory:
+            for signal in snapshot.model.render_signals(
+                start=start, end=end, execution_time=execution_time
+            ):
+                ready_intervals = signal_factory(signal).poll_ready_intervals()
+                if isinstance(ready_intervals, bool):
+                    if not ready_intervals:
+                        intervals = []
+                        break
+                elif isinstance(ready_intervals, list):
+                    intervals = [
```

Comment:
can this just be set to the intervals returned by the function?

Reply:
This was addressed in the new interface in https://github.com/TobikoData/sqlmesh/pull/2822/commits/e9d9119249d07679bc4c04c913d4563ffdf8aef7